src/Compat.jl

```diff
@@ -385,6 +385,58 @@ end
     end
 end
 
+# this function is available as of Julia 1.9
+# https://github.com/JuliaLang/julia/pull/45607
+@static if !isdefined(Base, :pkgversion)
+    using TOML: parsefile
+    export pkgversion
+
+    const project_names = ("JuliaProject.toml", "Project.toml")
+    const manifest_names = ("JuliaManifest.toml", "Manifest.toml")
+    const preferences_names = ("JuliaLocalPreferences.toml", "LocalPreferences.toml")
+
+    function locate_project_file(env::String)
+        for proj in project_names
+            project_file = joinpath(env, proj)
+            if Base.isfile_casesensitive(project_file)
+                return project_file
+            end
+        end
+        return true
```

Comment:
```suggestion
        return nothing
```
Would be more idiomatic, or is there a good reason for return `true` if the project file could not be found?

Reply:
IIRC I grabbed this from the Base implementation, but this isn't part of the public API, so I'm fine with `nothing`

Reply:
Reference to Base returning `true`: https://github.com/JuliaLang/julia/blob/0f7b598ad2a690cae4fe67b3f834e476a77399f4/base/loading.jl#L584. I also think it's fine either way.